newIDE/app/src/ResourcesLoader/index.js

```diff
@@ -146,4 +147,22 @@ export default class ResourcesLoader {
 
     return resourceName;
   }
+
+  /**
+   * Get the fully qualified URL/filename associated with the given resource.
+   */
+  static getResourceJsonData(project: gdProject, resourceName: string) {
+    if (!project.getResourcesManager().hasResource(resourceName))
+      return Promise.reject(); // TODO or something if resource does not exist
+
+    // Get the resource
+    const resource = project.getResourcesManager().getResource(resourceName);
+    if (resource.getKind() !== 'json') return Promise.reject(); // TODO something if resource is not json
+
+    // Get its URL (can be a path locally, but we still call it a URL)
+    const fullUrl = ResourcesLoader.getResourceFullUrl(project, resourceName);
```

Comment:
@4ian is this a correct way to gain access to getResourceFullUrl from here?

Reply:
I'm not sure I understand what you need exactly?

Reply:
because axios requires it?

Reply:
The reason I have 
static ResourcesLoader = ResourcesLoader;

is so I can access the get full url method here :( 
ResourcesLoader.getResourceFullUrl...

but you told me to remove that

Reply:
So two things:
- Using `ResourcesLoader.getResourceFullUrl` is totally fine.
- You almost certainly don't need `static ResourcesLoader = ResourcesLoader`... because it's not even in this file :)

Reply:
yup removed it now, but had to move getResourceJsonData in the process

Reply:
That's perfect, I was about to write that you should do this! `getResourceJsonData` is actually better in `PixiResourcesLoader`, as this is the class exposing the functions to load stuff in JsExtension.js

`ResourcesLoader` is a "low level", implementation detail that is just there to compute the URL/paths to resources. Arguably, the namings are a bit confusing.

Anyway, you found the right solution! :)